### PR TITLE
Adds a roomba

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

Cat roomba

## Why It's Good For The Game
It keeps prep not a fucking mess and reduces maptick from said mess

## Changelog
:cl:
code: claymores will no longer directly call bump
add: Adds a prep roomba
/:cl:

### CODE REVIEW

**Comment by neohenry** [comment]
inb4 roomba in WO

But finally, a way to clean prep.

**Comment by joookschad** [comment]
inb4 the roomba eats my armor plates as I buy them from the vendor

File: code/game/objects/machinery/roomba.dm [hunk 1]
```diff
@@ -0,0 +1,61 @@
+/// A cheap little roomba that runs around and keeps prep clean to decrease maptick and prep always being a fucking mess
+/obj/machinery/roomba
+	name = "nanotrasen roomba"
```
Comment (XSlayer300) [inline code/game/objects/machinery/roomba.dm:3]:
```suggestion
	name = "Nanotrasen roomba"
```

File: code/game/objects/machinery/roomba.dm [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	desc = "A nanotrasen roomba, designed to keep areas clean from dirty marines."
```
Comment (XSlayer300) [inline code/game/objects/machinery/roomba.dm:4]:
```suggestion
	desc = "A robot vacuum cleaner designed by Nanotrasen. The roomba is designed to keep areas clean from dirty marines."
```

**Comment by MLGTASTICa** [comment]
god please let us put a claymore ontop of it

**Comment by Chaznoodles** [comment]
Do the TG thing where we can tape knives to it and it stabs people in the leg/steals their hat and rank

**Comment by neohenry** [comment]
> 
> 
> Do the TG thing where we can tape knives to it and it stabs people in the leg/steals their hat and rank

bayonet roomba go brr

**Comment by Eskjjlj** [comment]
If only there was a good sound for it there could be a playsound call in the suck_item proc. Brrrrrrrrr every time it sucks

File: code/game/objects/machinery/roomba.dm [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+///Called when the roomba moves, sucks in all items held in the tile and sends them to cryo
+/obj/machinery/roomba/proc/suck_items()
+	SIGNAL_HANDLER
+	for(var/obj/item/sucker in loc)
+		sucker.store_in_cryo()
+		counter++
+
+/obj/machinery/roomba/attack_hand(mob/living/user)
+	. = ..()
+	visible_message("<span class='notice'>[user] lovingly pats the [src].</span>", "<span class='notice'>You lovingly pat the [src].</span>")
```
Comment (Hughgent) [inline code/game/objects/machinery/roomba.dm:61]: This should have multiple interactions depending on intent.

Make them feel the callousness of their soul if they click it with harm intent.

See `/obj/structure/shipmast/attack_hand(mob/living/user)` for an example.

**Review by Hughgent — APPROVED** [review]

**Comment by XSlayer300** [comment]
Okay, so can you make the roomba clean up objects ONLY on the marine prep? And make it undraggable to marines?

**Comment by Eskjjlj** [comment]
Warning: DEFCON 1. The Master Controller has still not fired within the last 80 ticks. Killing and restarting...
The Machines subsystem seems to be destabilizing the MC and will be offlined.
Happened after someone stucked the roomba

**Comment by Hughgent** [comment]
no more TMC until the infinite loop and MC issues are dealt with.

**Review by Hughgent — APPROVED** [review]
Another attempt, lets see if this destroys the MC this time.

**Comment by Eskjjlj** [comment]
According to Ruediger the roomba made the MC go to DEFCON1 until someone destroyed it.

**Comment by Hughgent** [comment]
image
comments or thoughts @TiviPlus

**Comment by Hughgent** [comment]
Secondary comment.

It seems that the 'claymore roomba' explodes upon "coming into contact with an object"
i.e. "It detonates upon moving to clean an object from the ground"

**Comment by terranaut1** [comment]
Unless there's an objective positive use and reason for marines to be able to put a claymore on it - and I don't mean it working as intended, but there being an actual reason for someone to do so that isn't memes or grief - just remove that bit entirely, please. All I've seen in admin chatter, and what I can personally guess is happening/going to happen from experience with SS13 in general, is negative, as far as a semi-randomly exploding thing goes.

File: code/game/objects/machinery/roomba.dm (hunk 3 above)
Comment (Hughgent) [reply code/game/objects/machinery/roomba.dm:79]: pending

**Comment by Hughgent** [comment]
I'll believe the doesn't crash the MC when I see it.

**Pull request merged**